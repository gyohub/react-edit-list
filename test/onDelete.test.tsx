import React from 'react';
import {
    cleanup,
    fireEvent,
    render,
    waitFor,
    waitForElementToBeRemoved
} from '@testing-library/react';

import ReactEditList, * as REL from 'react-edit-list';

import {schema, data, onLoad} from './data';

describe('onDelete()', () => {
    it('onDelete() undefined', async () => {
        let recvItem: REL.Row = {};
        const onDelete = jest.fn((item: REL.Row): Promise<void> => {
            recvItem = item;
            return Promise.resolve(undefined);
        });
        const r = render(<ReactEditList schema={schema} onLoad={onLoad} onDelete={onDelete} />);
        await waitFor(() => expect(r.getByText(/Desk/)).toBeInTheDocument());
        expect(r.container.innerHTML).toMatchSnapshot();

        fireEvent(r.getAllByText(/^x$/)[0], new MouseEvent('click', {bubbles: true}));
        await waitForElementToBeRemoved(() => r.queryByText('Desk'));
        expect(r.container.innerHTML).toMatchSnapshot();

        expect(onDelete).toBeCalledTimes(1);
        expect(recvItem.product).toBe('Desk');
        expect(recvItem.type).toBe('1');
        expect(recvItem.price).toBe(100);
        expect(recvItem.stock).toBe(20);
        r.unmount();
    });

    it('onDelete() false', async () => {
        let recvItem: REL.Row = {};
        const onDelete = jest.fn((item: REL.Row): false => {
            recvItem = item;
            return false;
        });
        const r = render(<ReactEditList schema={schema} onLoad={onLoad} onDelete={onDelete} />);
        await waitFor(() => expect(r.getByText(/Desk/)).toBeInTheDocument());
        expect(r.container.innerHTML).toMatchSnapshot();

        fireEvent(r.getAllByText(/^x$/)[0], new MouseEvent('click', {bubbles: true}));
        await waitFor(() => expect(onDelete).toBeCalledTimes(1));
        await waitFor(() => expect(r.getByText(/Desk/)).toBeInTheDocument());
        expect(r.container.innerHTML).toMatchSnapshot();

        expect(recvItem.product).toBe('Desk');
        expect(recvItem.type).toBe('1');
        expect(recvItem.price).toBe(100);
        expect(recvItem.stock).toBe(20);
        r.unmount();
    });

    it('disableDelete', async () => {
        const r = render(<ReactEditList schema={schema} onLoad={onLoad} disableDelete={true} />);
        await waitFor(() => expect(r.getByText(/Desk/)).toBeInTheDocument());
        expect(r.container.innerHTML).toMatchSnapshot();
        expect(r.queryAllByText(/^x$/)).toHaveLength(0);
        r.unmount();
    });
});
