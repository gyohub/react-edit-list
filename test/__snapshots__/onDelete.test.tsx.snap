// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onDelete() disableDelete 1`] = `"<table><thead><tr><th>product</th><th>type</th><th>price</th><th>stock</th><th></th></tr></thead><tbody><tr><td>Desk</td><td>capex</td><td>100</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>Printer</td><td>capex</td><td>500</td><td>10</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>Paper</td><td>consumable</td><td>5</td><td>2000</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>Chair</td><td>capex</td><td>50</td><td>50</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>Computer</td><td>capex</td><td>1000</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>Rent</td><td></td><td>2000</td><td></td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr></tbody></table>"`;

exports[`onDelete() onDelete() false 1`] = `"<table><thead><tr><th>product</th><th>type</th><th>price</th><th>stock</th><th></th></tr></thead><tbody><tr><td>Desk</td><td>capex</td><td>100</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Printer</td><td>capex</td><td>500</td><td>10</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Paper</td><td>consumable</td><td>5</td><td>2000</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Chair</td><td>capex</td><td>50</td><td>50</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Computer</td><td>capex</td><td>1000</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Rent</td><td></td><td>2000</td><td></td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr></tbody></table>"`;

exports[`onDelete() onDelete() false 2`] = `"<table><thead><tr><th>product</th><th>type</th><th>price</th><th>stock</th><th></th></tr></thead><tbody><tr><td>Desk</td><td>capex</td><td>100</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Printer</td><td>capex</td><td>500</td><td>10</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Paper</td><td>consumable</td><td>5</td><td>2000</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Chair</td><td>capex</td><td>50</td><td>50</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Computer</td><td>capex</td><td>1000</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Rent</td><td></td><td>2000</td><td></td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr></tbody></table>"`;

exports[`onDelete() onDelete() undefined 1`] = `"<table><thead><tr><th>product</th><th>type</th><th>price</th><th>stock</th><th></th></tr></thead><tbody><tr><td>Desk</td><td>capex</td><td>100</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Printer</td><td>capex</td><td>500</td><td>10</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Paper</td><td>consumable</td><td>5</td><td>2000</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Chair</td><td>capex</td><td>50</td><td>50</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Computer</td><td>capex</td><td>1000</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Rent</td><td></td><td>2000</td><td></td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr></tbody></table>"`;

exports[`onDelete() onDelete() undefined 2`] = `"<table><thead><tr><th>product</th><th>type</th><th>price</th><th>stock</th><th></th></tr></thead><tbody><tr><td>Printer</td><td>capex</td><td>500</td><td>10</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Paper</td><td>consumable</td><td>5</td><td>2000</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Chair</td><td>capex</td><td>50</td><td>50</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Computer</td><td>capex</td><td>1000</td><td>20</td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>Rent</td><td></td><td>2000</td><td></td><td><div style=\\"display: flex; flex-direction: row;\\"><button>x</button></div></td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td><div style=\\"display: flex; flex-direction: row;\\"></div></td></tr></tbody></table>"`;
